edge/registry/server.go

```diff
@@ -0,0 +1,114 @@
+package registry
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/gorilla/mux"
+	"github.com/portainer/agent"
+	"github.com/portainer/agent/edge"
+	"github.com/portainer/libhttp/request"
+	"github.com/portainer/libhttp/response"
+
+	httperror "github.com/portainer/libhttp/error"
+)
+
+type Handler struct {
+	*mux.Router
+	EdgeManager *edge.Manager
+}
+
+func NewEdgeRegistryHandler(edgeManager *edge.Manager) *Handler {
+	h := &Handler{
+		Router:      mux.NewRouter(),
+		EdgeManager: edgeManager,
+	}
+
+	h.Handle("/lookup", httperror.LoggerHandler(h.LookupHandler)).Methods(http.MethodGet)
+	return h
+}
+
+func (handler *Handler) LookupHandler(rw http.ResponseWriter, r *http.Request) *httperror.HandlerError {
+	stackManager := handler.EdgeManager.GetStackManager()
+	if stackManager == nil {
+		return &httperror.HandlerError{http.StatusInternalServerError, "Unable to retrieve stack manager", errors.New("Stack manager is not available")}
+	}
+
+	serverUrl, _ := request.RetrieveQueryParameter(r, "serverurl", false)
```

Comment:
should we skip running below code if serverUrl is ""

Reply:
Possibly, but the response needs to be empty credentials.  If not, you may not be able to deploy a public dockerhub stack.